src/mds/CInode.cc

```diff
@@ -3765,14 +3757,23 @@ void CInode::validate_disk_state(CInode::validated_data *results,
       // set up basic result reporting and make sure we got the data
       results->performed_validation = true; // at least, some of it!
       results->backtrace.checked = true;
-      results->backtrace.passed = false; // we'll set it true if we make it
+
+      int64_t pool;
+      if (in->is_dir())
+        pool = in->mdcache->mds->mdsmap->get_metadata_pool();
+      else
+        pool = in->inode.layout.fl_pg_pool;
+      inode_backtrace_t& memory_backtrace = results->backtrace.memory_value;
+      in->build_backtrace(pool, memory_backtrace);
+      bool equivalent, divergent;
+      int memory_newer;
```

Comment:
On Sat, Jan 23, 2016 at 10:34 AM, Gregory Farnum <notifications@github.com>
wrote:

> In src/mds/CInode.cc
> <https://github.com/ceph/ceph/pull/6682#discussion_r50614157>:
>
> > @@ -3765,14 +3757,23 @@ void CInode::validate_disk_state(CInode::validated_data *results,
> >        // set up basic result reporting and make sure we got the data
> >        results->performed_validation = true; // at least, some of it!
> >        results->backtrace.checked = true;
> > -      results->backtrace.passed = false; // we'll set it true if we make it
> > +
> > +      int64_t pool;
> > +      if (in->is_dir())
> > +        pool = in->mdcache->mds->mdsmap->get_metadata_pool();
> > +      else
> > +        pool = in->inode.layout.fl_pg_pool;
> > +      inode_backtrace_t& memory_backtrace = results->backtrace.memory_value;
> > +      in->build_backtrace(pool, memory_backtrace);
> > +      bool equivalent, divergent;
> > +      int memory_newer;
>
> There's really no point in pre-declaring equivalent, divergent, and
> memory_newer.
>

goto is used in this function


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/ceph/pull/6682/files#r50614157>.
>